### PR TITLE
This adds orderitem validation condition on quantity to only fire if the

### PR DESCRIPTION
**Pull request by ian-a-hickey**
order is not placed or if the quantity is being changed. This fixes an issue in a custom project.

---
This change is [Reviewable](https://reviewable.io/reviews/ten24/slatwall/5456)

### CODE REVIEW

File: model/entity/OrderItem.cfc [hunk 1]
```diff
@@ -469,6 +483,9 @@ component entityname="SlatwallOrderItem" table="SwOrderItem" persistent="true" a
 	}
 
 	public void function setQuantity(required numeric quantity){
+		if (isNull(variables.quantity) || arguments.quantity != variables.quantity){
```
Comment (rmarchandTen24) [inline model/entity/OrderItem.cfc]: you should just add a default=0 on the property so we won't have to check structkeyexists or isNull on variables.quantity

Comment (rmarchandTen24) [inline model/entity/OrderItem.cfc]: I think that the isNull check is only going to work if getQuantity was called before setQuantity.
structkeyexists would be used before going to the database
isNull for after

Comment (ian-a-hickey) [inline model/entity/OrderItem.cfc]: This is updated.

File: model/entity/OrderItem.cfc [hunk 2]
```diff
@@ -48,7 +48,7 @@ component entityname="SlatwallOrderItem" table="SwOrderItem" persistent="true" a
 	property name="price" ormtype="big_decimal";
 	property name="skuPrice" ormtype="big_decimal";
 	property name="currencyCode" ormtype="string" length="3";
-	property name="quantity" hb_populateEnabled="public" ormtype="integer";
+	property name="quantity" hb_populateEnabled="public" ormtype="integer" default="0";
```
Comment (rmarchandTen24) [inline model/entity/OrderItem.cfc]: Sorry for the confusion. Add the default to the non persistent property quantity has changed and not quantity itself

**Pull request merged**